### DISCUSSION TITLE
Example for using adapter plugin

### DISCUSSION THREAD

**Discussion by Srheft** [eradiate/eradiate#349] [Q&A] [answered]
Hi Eradiate,

A basic test question:
I was trying to verify radiance similarity at a given VZA and VZA+180deg in isotropic scenario but VZA+180yields zero radiation which prompted me to want to try "[twosided](https://eradiate.readthedocs.io/en/latest/rst/reference_plugins/generated/plugins/bsdfs/rpv.html?highlight=twosided#rahman-pinty-verstraete-reflection-model-rpv)" option but I'm not sure how to incorporate that. Is there an example?

Thanks,
S.

**Comment by leroyvn** [comment]
Hi @Srheft, I am not sure if I understand what you are trying to achieve. Eradiate uses the [ISO 80000-2:2019](https://eradiate.readthedocs.io/en/v0.23.1/rst/user_guide/conventions.html#spherical-coordinates) convention to define spherical coordinates. Consequently, the zenith angle is in the $[0, 180]°$ interval, and trying and compare radiance values for VZA and VZA+180° is a scenario which should raise an error. Note that the recently merged #345 fixes insufficient checks in measure initialization and prevents the use of illegal zenith angle values.

**Comment by Srheft** [comment]
OK, thanks. This was meant to simply demonstrate the isotropic nature of the radiance at paired angles. And I thought "twosided" option expands the valid angle range but it was not related.

**Reply by leroyvn** [reply, marked as answer]
Ok, I think I see what you mean. Legal zenith values must be in [0, 180]°; we could have a rule to map values outside of that range to sensible directions—we do for azimuths outside of [0, 360]°—but we haven't implemented this yet. We interpret negative zeniths for some measure constructors, but not in a generic way. Generalizing these rules is however not a high priority on our list a.t.m.

What `twosided` does is that it assigns a nested Mitsuba single-face BSDF plugin to the back face of a shape—which would otherwise look black. So in practice, if you have a Lambertian surface, using `twosided` on it will make both sides (zenith in [0, 90]° and zenith in [90, 180]°) Lambertian.
